new-line-before-expect fails when returning or awaiting an expect

<!-- Fill your answers in between -->
**Do you want to request a *feature* or report a *bug*?**
Bug
**What is the current behavior?**
Returning and awaiting an expect results in a rule failure

**If the current behavior is a bug, please provide the steps to reproduce.**
Here are some examples:
```
test('resolves to lemon', () => {
  return expect(Promise.resolve('lemon')).resolves.toBe('lemon');
});
```
```
test('resolves to lemon', async () => {
  await expect(Promise.resolve('lemon')).resolves.toBe('lemon');
  await expect(Promise.resolve('lemon')).resolves.not.toBe('octopus');
});
```

Both from [Jest's docs](https://facebook.github.io/jest/docs/en/expect.html#resolves).
**What is the expected behavior?**
Return/awaiting expects should not fail
**Please mention your node.js, eslint-plugin-jasmine and operating system version.**
node: v8.9.0
eslint-plugin-jasmine: 2.9.3
os: 10.12.6

@kaicataldo it looks like this:

```
> npx eslint .

/my-file.js
  45:15  error  No new line before expect  jasmine/new-line-before-expect

✖ 1 problem (1 error, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```

--- a/docs/rules/new-line-before-expect.md
+++ b/docs/rules/new-line-before-expect.md
@@ -51,6 +51,26 @@ describe("", function() {
 
 ```js
 describe("", function() {
+  it("", async function() {
+    var a = 1;
+
+    await expect(a).toBe(1);
+  });
+});
+```
+
+```js
+describe("", function() {
+  it("", function() {
+    var a = 1;
+
+    return expect(a).toBe(1);
+  });
+});
+```
+
+```js
+describe("", function() {
   it("", function() {
     var a = 1;
 

--- a/lib/rules/new-line-before-expect.js
+++ b/lib/rules/new-line-before-expect.js
@@ -26,7 +26,11 @@ module.exports = function (context) {
         }
         lastExpectNode = node
         var sourceCode = context.getSourceCode()
-        const prevToken = sourceCode.getTokenBefore(node)
+        let prevToken = sourceCode.getTokenBefore(node)
+        if (prevToken.value === 'await' || prevToken.value === 'return') {
+          node = prevToken
+          prevToken = sourceCode.getTokenBefore(prevToken)
+        }
         if (prevToken) {
           if (prevToken.type === 'Punctuator' && prevToken.value === '{') {
             return

--- a/test/rules/new-line-before-expect.js
+++ b/test/rules/new-line-before-expect.js
@@ -3,8 +3,10 @@
 var rule = require('../../lib/rules/new-line-before-expect')
 var linesToCode = require('../helpers/lines_to_code')
 var RuleTester = require('eslint').RuleTester
-
-var eslintTester = new RuleTester()
+const parserOptions = {
+  ecmaVersion: 8
+}
+var eslintTester = new RuleTester({ parserOptions })
 
 eslintTester.run('new line before expect', rule, {
   valid: [
@@ -41,6 +43,15 @@ eslintTester.run('new line before expect', rule, {
       '});'
     ]),
     linesToCode([
+      'describe("", function() {',
+      ' it("", async function(){',
+      '  var a = 1',
+      '',
+      '  await expect(a).toBe(1)',
+      ' });',
+      '});'
+    ]),
+    linesToCode([
       'it("", helper(function() {',
       '  var a = 1',
       '',
@@ -69,6 +80,11 @@ eslintTester.run('new line before expect', rule, {
       ' ',
       '  expect(1).toEqual(1);',
       '}));'
+    ]),
+    linesToCode([
+      'it("", function() {',
+      '  return expect(1).toEqual(1);',
+      '});'
     ])
 
   ],
@@ -114,6 +130,46 @@ eslintTester.run('new line before expect', rule, {
         '  var a = 1',
         '',
         '  expect(a).toEqual(1);',
+        '}));'
+      ])
+    },
+    {
+      code: linesToCode([
+        'it("", helper(async function() {',
+        '  var a = 1',
+        '  await expect(a).toEqual(1);',
+        '}));'
+      ]),
+      errors: [
+        {
+          message: 'No new line before expect'
+        }
+      ],
+      output: linesToCode([
+        'it("", helper(async function() {',
+        '  var a = 1',
+        '',
+        '  await expect(a).toEqual(1);',
+        '}));'
+      ])
+    },
+    {
+      code: linesToCode([
+        'it("", helper(function() {',
+        '  var a = 1',
+        '  return expect(a).toEqual(1);',
+        '}));'
+      ]),
+      errors: [
+        {
+          message: 'No new line before expect'
+        }
+      ],
+      output: linesToCode([
+        'it("", helper(function() {',
+        '  var a = 1',
+        '',
+        '  return expect(a).toEqual(1);',
         '}));'
       ])
     },

